docs/developer/release/3-update-version-in-code.md

```diff
@@ -0,0 +1,67 @@
+# Update Version in Code
+
+The project must be updated to reference the upcoming release tag whenever a new release is being made.
+
+## Release Version
+
+The release version prefix was previously determined in [Create Release Branch](./1-create-release-branch.md). The `[release version]` is defined as follows.
+
+- Release Candidate Version (RCV)
+
+    - The RCV will look like `[release version prefix].x-rc.y`.
+    - For example, `v0.31.0-rc.0` is the first RCV for the v0.31.0 release.
+
+- Stable Release Version (SRV)
+
+    - The SRV will look like `[release version prefix].0`.
+    - For example, `v0.31.0` is the SRV for the v0.31.0 release.
+
+- Patch Release Version (PRV)
+
+    - The PRV will look like `[release version prefix].x`.
+    - For example, `v0.31.1` is the first PRV for the v0.31.0 release.
+
+*NOTE: This value will be referred to as `[release version]` in this documentation*
+
+## Steps
+
+1. Create a branch from `main` for [grafana/agent](https://github.com/grafana/agent).
+
+2. Update the `CHANGELOG.md`.
+
+    1. `CHANGELOG.md` Header
+        - First RCV or a PRV
+            - Add a new header under `Main (unreleased)` for `[release version] (YYYY-MM-DD)`.
+        - Additional RCV or SRV
+            - Update the header `[previous RCV] (YYYY-MM-DD)` to `[release version] (YYYY-MM-DD)`. The date may need updating.
+
+    2. Move the unreleased changes we want to add to the release branch from `Main (unreleased)` to `[release version] (YYYY-MM-DD)`.
+
+    3. Update appropriate places in the codebase that have the previous version with the new version determined above.
+
+        *NOTE: Please do not update the operations/helm directory. It is updated independently from Agent releases for now.*
+    
+        *NOTE: This will require some tribal knowledge not documented here (yet).*
```

Comment:
Timeless documentation: 

```suggestion
    3. Update appropriate places in the codebase that have the previous version with the new version determined above.

        * Do **not** update the `operations/helm` directory. It is updated independently from Agent releases.
```

We can update the instructions if this ever changes, but we don't need to document that it may change.